lib/util/AppendOnlyStackedSet.js

```diff
@@ -0,0 +1,51 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+/**
+ * @template T
+ */
+class AppendOnlyStackedSet {
+	constructor(sets = []) {
+		this._sets = sets;
+		this._current = undefined;
+	}
```

Comment:
I am also working on strict types for webpack right now, so now when I added something new I enable `strict` in `tsconfig.json` and put write all required types, before we have 6500+ ts error, now it is around 2500, I want to reduce them more, so let's make a better types here too